drivers/dac/ad5754r/ad5754r.c

```diff
@@ -0,0 +1,684 @@
+ * @param dev - The device structure.
+ * @param instr_addr - The instruction address comprising register and
+ *				dac channel address.
+ * @param mask - Specific bits mask.
+ * @param reg_val - The data to be written.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_update_bits(struct ad5754r_dev *dev,
+			uint8_t instr_addr,
+			uint16_t mask,
+			uint16_t reg_val)
+{
+	int ret;
+	uint16_t reg_data;
+
+	if (!dev)
+		return -EINVAL;
+
+	ret = ad5754r_read(dev, instr_addr, &reg_data);
+	if (ret)
+		return ret;
+
+	reg_data &= ~mask;
+	reg_data |= reg_val;
+
+	return ad5754r_write(dev, instr_addr, reg_data);
+}
+
+/**
+ * @brief Read specific register bits using supplied mask.
+ * @param dev - The device structure.
+ * @param instr_addr - The instruction address comprising register and
+ *				dac channel address.
+ * @param mask - Specific bits mask.
+ * @param reg_val - The data read from the register.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_read_masked(struct ad5754r_dev *dev,
+			uint8_t instr_addr,
+			uint16_t mask,
+			uint16_t *reg_val)
+{
+	int ret;
+	uint16_t reg_data;
+
+	if (!dev)
+		return -EINVAL;
+
+	ret = ad5754r_read(dev, instr_addr, &reg_data);
+	if (ret)
+		return ret;
+
+	*reg_val = no_os_field_get(mask, reg_data);
+
+	return 0;
+}
+
+/**
+ * @brief Update DAC register value for specific channel.
+ * @param dev - The device structure.
+ * @param chn - The index of channel being addressed.
+ * @param value - The data to be written to the dac channel.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_update_dac_ch_register(struct ad5754r_dev *dev,
+				   uint8_t chn,
+				   uint16_t value)
+{
+	if (!dev || (chn > AD5754R_NUM_CHANNELS))
+		return -EINVAL;
+
+	return ad5754r_write(dev, AD5754R_PREPARE_INSTR_ADDR(AD5754R_REG_DAC,
+			     chn),
+			     value);
+}
+
+/**
+ * @brief Update DAC register value for all channels.
+ * @param dev - The device structure.
+ * @param value - The data to be written to the dac channel registers.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_update_dac_all_ch_registers(struct ad5754r_dev *dev,
+					uint16_t value)
+{
+	if (!dev)
+		return -EINVAL;
+
+	return ad5754r_write(dev,
+			     AD5754R_PREPARE_INSTR_ADDR(AD5754R_REG_DAC,
+					     AD5754R_DAC_CH_ALL_ADDR),
+			     value);
+}
+
+/**
+ * @brief Read DAC register value for specific channel.
+ * @param dev - The device structure.
+ * @param chn - The index of channel being addressed.
+ * @param value - The data read from dac channel register.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad5754r_read_dac_ch_register(struct ad5754r_dev *dev,
+				 uint8_t chn,
+				 uint16_t *value)
+{
+	if (!dev || !value || (chn >= AD5754R_NUM_CHANNELS))
+		return -EINVAL;
+
+	return ad5754r_read(dev,
+			    AD5754R_PREPARE_INSTR_ADDR(AD5754R_REG_DAC, chn),
```

Comment:
nit: i'd shot a bit the name of this macro `AD5754R_PREPARE_INSTR_ADDR`. this might fix multiple odd indentations throughout the code.